### PR TITLE
Update paths in docs

### LINKED ISSUE #12340: Update docs for new src layout
Following the changes in #12338, we need to update the paths in the contributor docs. One location I noticed is in python.rst, for example.

### PR DESCRIPTION
**Pull request by tcmetzger**
Update paths in docs following https://github.com/bokeh/bokeh/pull/12338.

Fixes #12340

### CODE REVIEW

File: docs/bokeh/source/docs/dev_guide/writing_tests.rst [hunk 1]
```diff
@@ -87,7 +87,7 @@ Python integration tests use pytest fixtures to handle the web driver
 configuration and interaction with Selenium. Depending on which context you
 want to test an object in, choose from ``bokeh_model_page``,
 ``single_plot_page``, or ``bokeh_server_page``. See
-:bokeh-tree:`bokeh/_testing/plugins/project.py` for more details.
```
Comment (tcmetzger) [inline docs/bokeh/source/docs/dev_guide/writing_tests.rst:90]: @bryevdv This path seems to have changed quite a bit - did I reference the correct file here?

Comment (bryevdv) [reply docs/bokeh/source/docs/dev_guide/writing_tests.rst:90]: these were moved out of the package to the test dir: https://github.com/bokeh/bokeh/tree/branch-3.0/tests/support/plugins

**Comment by bryevdv** [comment]
FYI I think the examples readme still has a link to `file`

https://github.com/bokeh/bokeh/blob/branch-3.0/examples/plotting/README.md

**Review by bryevdv — APPROVED** [review]

**Comment by tcmetzger** [comment]
@bryevdv wow, good eye! 😀 I have updated that file as well!

**Pull request merged**